x-pack/plugins/enterprise_search/server/routes/enterprise_search/engines.ts

```diff
@@ -59,11 +61,13 @@ export function registerEnginesRoutes({ log, router }: RouteDependencies) {
     },
     elasticsearchErrorHandler(log, async (context, request, response) => {
       const { client } = (await context.core).elasticsearch;
-      const engine = await client.asCurrentUser.transport.request<EnterpriseSearchEngine>({
+      const engine = await client.asCurrentUser.transport.request<EnterpriseSearchEngineDetails>({
```

Comment:
This should stay `EnterpriseSearchEngine` that is what we get from elasticsearch. But we return `EnterpriseSearchEngineDetails` from this handler.